src/test/groovy/com/rabbitmq/http/client/ClientSpec.groovy

```diff
@@ -845,6 +901,73 @@ class ClientSpec extends Specification {
     client << clients()
   }
 
+  @Unroll
+  def "GET /api/queues with paging and details"() {
+    given: "at least one queue was declared"
+    Connection conn = cf.newConnection()
+    Channel ch = conn.createChannel()
+    def queues = (0..15).collect {it ->
+      def q = "queue-for-paging-and-details-test-" + it
+      ch.queueDeclare(q, false, false, false, null)
+      ch.queueBind(q, "amq.fanout", "")
+      q
+    }
+
+    ExecutorService executorService = Executors.newSingleThreadExecutor()
+    def publishing = {
+      while (!Thread.currentThread().isInterrupted()) {
+        ch.basicPublish("amq.fanout", "", null, "".getBytes(StandardCharsets.UTF_8))
+        try {
+          Thread.sleep(10)
+        } catch (InterruptedException e) {
+          return
+        }
+      }
+    }
+    executorService.submit(publishing)
+
+    when: "client lists queues with details"
+    def queryParameters = new DetailsParameters(new QueryParameters()
```

Comment:
Based on the suggestion made above this could would look like this
```
def detailsParameters = new DetailsParameters()
             .messageRates(60, 5)
            .lengths(60, 5)
detailsParameters.queryParameters()
            .name("^queue-for-paging-and-details-test-*", true)
            .pagination()
            .pageSize(10)

def request = { client.getQueues(detailsParameters) }
```